sigstore/_cli.py

```diff
@@ -143,13 +143,18 @@ def _verify(files, certificate_path, signature_path, cert_email):
     logger.debug(f"Using signature from: {signature_path.name}")
     signature = signature_path.read()
 
+    verified = True
     for file in files:
-        click.echo(
-            verify(
-                file=file,
-                certificate=certificate,
-                signature=signature,
-                cert_email=cert_email,
-                output=click.echo,
-            )
-        )
+        if verify(
+            file=file,
+            certificate=certificate,
+            signature=signature,
+            cert_email=cert_email,
+        ) is not None:
```

Comment:
I think we should probably do something slightly more here than check if it's not `None`. I realize that `VerificationResult`s are currently just a stub, but maybe we can add `VerificationResult.success` or `.verified` or even make it truthy/falsey so we can always return a result and do something like `if all(results):`?

Reply:
Yeah, I'm of two minds on this: it would be nice for `VerificationResult` to be `__bool__` so that we could have a description of why verification failed, but the current failure modes are basically descriptionless (signature failure is a pure function of the inputs, and Rekor failure is essentially just "we couldn't find a record"). OTOH the success modes have _some_ possible describable state (such as a dump of the Rekor entry/entries).

But yeah, `is not None` is not very descriptive. I'll change it to be `__bool__` and make it an unconditional return.

Reply:
[febfae7](https://github.com/sigstore/sigstore-python/pull/71/commits/febfae7cbaa2ceaa19248678c54f89803a276ca9)